GetBPDAwards/handler.ts

```diff
@@ -0,0 +1,185 @@
+                  cit_rank_insert_date: award.cit_rank_insert_date?.toISOString(),
+                  cit_rank_update_date: award.cit_rank_update_date?.toISOString(),
+                  ranking_date: award.ranking_date?.toISOString()
+                }
+              ],
+              fiscal_code: award.fiscal_code
+            } as AwardsList);
+      }
+      return {
+        ...acc,
+        awards: [
+          ...acc.awards,
+          {
+            ...award,
+            aw_per_aw_period_end: award.aw_per_aw_period_end.toISOString(),
+            aw_per_aw_period_start: award.aw_per_aw_period_start.toISOString(),
+            aw_per_insert_date: award.aw_per_insert_date?.toISOString(),
+            aw_per_update_date: award.aw_per_update_date?.toISOString(),
+            aw_winn_aw_period_end: award.aw_winn_aw_period_end?.toISOString(),
+            aw_winn_aw_period_start: award.aw_winn_aw_period_start?.toISOString(),
+            aw_winn_insert_date: award.aw_winn_insert_date?.toISOString(),
+            aw_winn_update_date: award.aw_winn_update_date?.toISOString(),
+            cit_rank_insert_date: award.cit_rank_insert_date?.toISOString(),
+            cit_rank_update_date: award.cit_rank_update_date?.toISOString(),
+            ranking_date: award.ranking_date?.toISOString()
+          }
+        ]
+      } as AwardsList;
+    }, undefined)
+  );
+};
+
+type ResponseErrorTypes =
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorInternal
+  | IResponseErrorValidation;
+
+type IHttpHandler = (
+  context: Context,
+  userAndCitizenId: RequestCitizenToAdUserAndFiscalCode
+) => Promise<IResponseSuccessJson<AwardsList> | ResponseErrorTypes>;
+
+export function GetBPDAwardsHandler(
+  awardRepository: TaskEither<Error, Repository<Award>>
+): IHttpHandler {
+  return async (context, userAndfiscalCode) => {
+    return awardRepository
+      .chain(award =>
+        tryCatch(
+          () => award.find({ fiscal_code: userAndfiscalCode.fiscalCode }),
+          (err: unknown) => {
+            context.log.error(
+              `GetBPDAwardsHandler|ERROR|Find awards query error [${err}]`
+            );
+            return new Error("Award find query error");
+          }
+        )
+      )
+      .mapLeft<
+        | IResponseErrorInternal
+        | IResponseErrorNotFound
+        | IResponseErrorValidation
+      >(err => ResponseErrorInternal(err.message))
+      .chain(
+        fromPredicate(
+          awardData => awardData.length > 0,
+          () => ResponseErrorNotFound("Not found", "Awards not found")
+        )
+      )
+      .chain(awardData =>
+        fromEither(toApiBPDAward(awardData)).mapLeft(err =>
+          ResponseErrorValidation(
+            "Invalid AwardsList object",
+            readableReport(err)
+          )
+        )
+      )
+      .map(ResponseSuccessJson)
+      .fold<ResponseErrorTypes | IResponseSuccessJson<AwardsList>>(
```

Comment:
Maybe we can map `ResponseSuccessJson` directly into `fold`.

Reply:
Fixed in a295421